open_seq2seq/encoders/convs2s_encoder.py

```diff
@@ -0,0 +1,201 @@
+# Copyright (c) 2018 NVIDIA Corporation
+"""
+Conv-based encoder
+"""
+from __future__ import absolute_import, division, print_function
+from __future__ import unicode_literals
+
+import tensorflow as tf
+from .encoder import Encoder
+
+from open_seq2seq.parts.convs2s import embedding_layer
+from open_seq2seq.parts.convs2s.utils import get_padding_bias, get_padding
+from open_seq2seq.parts.convs2s import ffn_wn_layer, conv_wn_layer
+
+# Default value used if max_input_length is not given
+MAX_INPUT_LENGTH = 100
+
+
+class ConvS2SEncoder(Encoder):
+  """
+  Fully convolutional Encoder of ConvS2S
+  """
+  @staticmethod
+  def get_required_params():
+    return dict(Encoder.get_required_params(), **{
+      "encoder_layers": int,
+
+      "src_emb_size": int,
+      "src_vocab_size": int,
+      "pad_embeddings_2_eight": bool,
+
+      "embedding_dropout_keep_prob": float,
+      "hidden_dropout_keep_prob": float,
+    })
+
+  @staticmethod
+  def get_optional_params():
+    return dict(Encoder.get_optional_params(), **{
+      "conv_knum": list,
+      "conv_kwidth": list,
+
+      "att_layer_num": int,
+
+      'max_input_length': int,
+      'PAD_SYMBOL': int,
+
+      'mask_paddings': bool
+    })
+
+  def __init__(self, params, model, name="convs2s_encoder_with_emb", mode='train'):
+    """
+    Initializes convolutional encoder with embeddings
+    :param params: dictionary with encoder parameters
+    Must define:
+      * src_vocab_size - data vocabulary size
+      * src_emb_size - size of embedding to use
+      * mode - train or infer
+    """
+    super(ConvS2SEncoder, self).__init__(params, model, name=name, mode=mode)
+
+    self._src_vocab_size = self.params['src_vocab_size']
+    self._src_emb_size = self.params['src_emb_size']
+    self.layers = []
+    self._mode = mode
+
+
+  def _encode(self, input_dict):
+    """
+    Encodes data into representation
```

Comment:
This documentation format looks wrong. It should have "Args:" and (optionally) "Returns" sections. 
See ``open_seq2seq/encoders/encoder.py:151`` for example